src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -299,565 +354,456 @@ void InferRequestBase::changeDefaultPtr() {
-
-                _inputs[name] = make_blob_with_precision(desc);
-                _inputs[name]->allocate();
+                tensor = ov::Tensor(port.get_element_type(), tensor_shape);
+                ov::ISyncInferRequest::set_tensor(port, tensor);
 
+                auto desc = create_tensor_desc(tensor);
                 if (!isDynamic &&
-                    desc == MemoryDescUtils::convertToTensorDesc(graph->getInputNodeByName(name)->getChildEdgesAtPort(0)[0]->getMemory().getDesc()) &&
-                        graph->_normalizePreprocMap.find(name) == graph->_normalizePreprocMap.end()) {
-                    externalPtr[name] = _inputs[name]->buffer();
+                    desc == MemoryDescUtils::convertToTensorDesc(
+                                graph->getInputNodeByName(name)->getChildEdgesAtPort(0)[0]->getMemory().getDesc()) &&
+                    graph->_normalizePreprocMap.find(name) == graph->_normalizePreprocMap.end()) {
+                    external_ptr[name] = tensor.data();
                 }
-            } else {
-                IE_THROW() << "Blob with name: " << name << " exists in CPU plugin graph, but absents in network inputs";
             }
+        } else {
+            OPENVINO_THROW("Tensor with name: ", name, " exists in CPU plugin graph, but absents in network inputs");
         }
-        data = _inputs[name];
     }
 
-    const auto &outMap = graph->outputNodesMap;
+    const auto& outMap = graph->outputNodesMap;
     auto output = outMap.find(name);
     if (output != outMap.end()) {
-        if (_outputs.find(name) == _outputs.end()) {
-            auto outputNode = modelOutputsMap.find(name);
-            if (modelOutputsMap.find(name) != modelOutputsMap.end()) {
-                const auto shape = outputNode->second->get_input_partial_shape(0);
-                bool isDynamic = shape.is_dynamic();
-
-                if (!data) {
-                    InferenceEngine::SizeVector dims;
-                    if (isDynamic) {
-                        dims = InferenceEngine::SizeVector(shape.rank().get_length(), 0);
-                    } else {
-                        dims = shape.to_shape();
-                    }
-
-                    InferenceEngine::TensorDesc desc(InferenceEngine::details::convertPrecision(outputNode->second->get_input_element_type(0)),
-                                                     dims, InferenceEngine::TensorDesc::getLayoutByRank(dims.size()));
-
-                    data = make_blob_with_precision(desc);
-                    data->allocate();
+        auto output_port = m_output_ports_map.find(name);
+        if (output_port != m_output_ports_map.end()) {
+            auto& port = output_port->second;
+            tensor = ov::ISyncInferRequest::get_tensor(port);
+            const auto shape = port.get_partial_shape();
+            const bool isDynamic = shape.is_dynamic();
+
+            if (!tensor) {
```

Comment:
When will we get into this if clause? Is it possible?

Reply:
Most of time will go to if clause, because this method is to initialize tensors for infer_request. You can see that it is empty tensor when construct ISyncInferRequest object, it need create real tensors in its inherited class, it is here.